.github/actions/setup-go-tip/action.yml

```
@@ -22,11 +22,12 @@ runs:
           echo "Failed to download. Retrying in $wait_time seconds..."
           sleep $wait_time
         done
+        echo "download_status=${success}" >> $GITHUB_OUTPUT
 
-        if [[ "$success" == false ]]; then
-          echo "Failed to download Go Tip after $retries attempts"
-          exit 1
-        fi
+    - name: Install Go Tip
+      if: steps.try-to-download-go-tip.outputs.download_status == 'true'
```

Nit: you could make the naming more concise:

```suggestion
      if: steps.download.outputs.success == 'true'
```